File: 1062-partition-array-into-three-parts-with-equal-sum/partition-array-into-three-parts-with-equal-sum.java
class Solution {
     public boolean canThreePartsEqualSum(int[] num) {
      int sum = Arrays.stream(num).sum();

      if(sum %3 != 0) return false;
       int tsum = sum/3, temp = 0, found = 0;

       for(int i=0; i<num.length;i++){
        temp += num[i];
        if(temp == tsum){
            temp = 0;
            found++;
        }
       }
        return found>=3 ? true : false;
    }  

}

File: 1062-partition-array-into-three-parts-with-equal-sum/partition-array-into-three-parts-with-equal-sum.cpp
class Solution {
public:
       bool canThreePartsEqualSum(vector<int>& arr) {
    int s = 0, n = arr.size();
    for (int i = 0; i < n; i++) {
        arr[i] += s; s = arr[i];
    }
    if (s % 3)
        return false;
    s /= 3;
    int c = 1;
    for (int i = 0; i < n; i++)
        if (arr[i] == s * c)
            c++;
    return c > 3;
}
};